crates/recent_projects/src/dev_servers.rs

```diff
@@ -254,45 +265,112 @@ impl DevServerProjects {
+    pub fn create_dev_server(&mut self, manual_setup: bool, cx: &mut ViewContext<Self>) {
         let name = get_text(&self.dev_server_name_input, cx);
         if name.is_empty() {
             return;
         }
 
-        let ssh_connection_string = if self.use_server_name_in_ssh == Selection::Selected {
-            Some(name.clone())
-        } else {
+        let ssh_connection_string = if manual_setup {
             None
+        } else {
+            Some(name.clone())
         };
 
         let dev_server = self.dev_server_store.update(cx, |store, cx| {
-            store.create_dev_server(name, ssh_connection_string, cx)
+            store.create_dev_server(name, ssh_connection_string.clone(), cx)
         });
 
+        let workspace = self.workspace.clone();
+
         cx.spawn(|this, mut cx| async move {
             let result = dev_server.await;
 
-            this.update(&mut cx, |this, cx| match &result {
+            match result {
                 Ok(dev_server) => {
-                    this.focus_handle.focus(cx);
-                    this.mode = Mode::CreateDevServer(CreateDevServer {
-                        creating: false,
-                        dev_server: Some(dev_server.clone()),
-                    });
-                }
-                Err(_) => {
-                    this.mode = Mode::CreateDevServer(Default::default());
+                    if let Some(ssh_connection_string) =  ssh_connection_string {
+                    if let Some(terminal_panel) = workspace
+                        .update(&mut cx, |workspace, cx| workspace.panel::<TerminalPanel>(cx))
+                        .ok()
+                        .flatten()
+                        .with_context(|| anyhow::anyhow!("No terminal panel"))
+                        .log_err()
+                    {
+                        let install_command = if env::var("ZED_RPC_URL") == Ok("http://localhost:8080/rpc".to_string()) {
+                            format!(r#"cd; (curl -sSL https://zed.dev/install.sh || wget -qO- https://zed.dev/install.sh) | bash && ZED_RPC_URL=http://localhost:8080/rpc ~/.local/bin/zed --dev-server-token {} && exec $SHELL"#, dev_server.access_token)
+                        } else {
+                            format!(r#"cd; (curl -sSL https://zed.dev/install.sh || wget -qO- https://zed.dev/install.sh) | bash && ~/.local/bin/zed --dev-server-token {}"#, dev_server.access_token)
+                        };
+
+                        let real_ssh = which::which("ssh")?;
+                        let tmp_dir = tempfile::tempdir()?;
+                        let path = tmp_dir.path().join("ssh");
+                        dbg!(&path);
+                        let mut file = File::create(path.clone())?;
+                        write!(&mut file, "#!/bin/bash\nexec {} \"$@\" -R 8080:localhost:8080 sh -c {}", real_ssh.to_string_lossy(), shlex::try_quote(&install_command)?)?;
```

Comment:
`/bin/bash`, while very common, is not a standard location (and will fail on certain distros such as NixOS, and possibly various BSDs). POSIX compliant options are `/bin/sh` and `/usr/bin/env bash`. If you can get away with it I recommend using `/bin/sh` since bash is technically not guaranteed to be installed (`/usr/bin/env` is effectively guaranteed to exist and know where bash is, if that is installed).

Reply:
related to this, if you felt like taking a pass at `install.sh`... I don't think we're doing anything too bash specific, but it'd be neat to have that run with just sh.

Reply:
I gave it a quick scan and I think it mainly needs to have the conditions of if statements converted to single brackets? (and string equality checks to a single `=`)

It would probably be nice to have a `trap "rm -rf $tmpdir" EXIT` to clean up the temp files after a successful install though. Maybe on error too, but on the other hand it is useful to have the files around for manual intervention in that case.